functions/webhooks/taskrouterCallback.protected.ts

```diff
@@ -87,10 +95,37 @@ export const handler = async (
   const resolve = bindResolve(callback)(response);
 
   try {
-    const { EventType: eventType } = event;
-
+    const { EventType: eventType, TaskSid } = event;
     const taskAttributes = JSON.parse(event.TaskAttributes!);
 
+    if (isTaskCreated(eventType)) {
+      const client = context.getTwilioClient();
+      const workplaceSid = context.TWILIO_WORKSPACE_SID;
+      const chatServiceSid = context.CHAT_SERVICE_SID;
+
+      if (TaskSid === undefined) {
+        resolve(error400('TaskSid'));
+        return;
+      }
+      const task = await client.taskrouter.workspaces(workplaceSid).tasks(TaskSid).fetch();
+
+      const { channelSid } = JSON.parse(task.attributes);
+      if (channelSid === undefined) {
+        resolve(error400('channelSid'));
+        return;
+      }
+
+      // Fetch channel to update with a taskId
+      const channel = await client.chat.services(chatServiceSid).channels(channelSid).fetch();
+
+      await channel.update({
+        attributes: JSON.stringify({
+          ...JSON.parse(channel.attributes),
+          taskSid: task.sid,
+        }),
+      });
+    }
+
     if (isCreateContactTask(eventType, taskAttributes)) {
```

Comment:
I believe that your changes could be inside this `if (isCreateContactTask(...))`, instead of the newly created `if (isTaskCreated(...))`.

The first one already checks that it is a task creation, and also checks if it's not an offline task. This latter check is good because offline tasks do not have a channel.

But I liked that your changes are isolated from the existing ones. It's up to you if you want to move your code into this existing if-clause or not. Maybe consult @GPaoloni if he's got any opinions here.